### PR TITLE
Bulk convert to using testify/suite for tests

### PR DESCRIPTION
**Pull request by ashb**
This is in some regards a "change for no purpose" but it lets us more
easily add setup or teardown to tests.

The `go test` compatible test was named after the Go packge
`./pkg/workspace` becomes `TestPkgWorkspaceSuite`.

This change was almost entirely automatically applied using
`github.com/uber-go/gopatch`

I have not converted 100% of tests to use suites, some of them proved to have ordering bugs (tests relying on state left over or setup from a previous test) that I wasn't able to easily/quickly solve, and I haven't applied Setup/Teardown functions universally, only where I specifically noticed them.

The patch file used was this:

```diff
@@
var foo identifier
@@
 import "testing"

-	t.Errorf(...)
+ s.Fail(...)

@@
@@
 import "github.com/stretchr/testify/assert"

-if err != nil {
-	t.Error(err)
-}
+assert.NoError(t, err)

@@
@@
 import "github.com/stretchr/testify/assert"

-assert.Nil(t, err)
+assert.NoError(t, err)

@@
var foo identifier
@@
-func foo(t *testing.T) {
+func (s *Suite) foo(t *testing.T) {
  ...
 }

@@
var name expression
@@
-t.Run(name, func(t *testing.T) {
+s.Run(name, func() {
  ...
 })

@@
var bar identifier
@@
 import "github.com/stretchr/testify/assert"

- assert.bar(t, ...)
+ s.bar(...)

@@
@@

-if err != nil {
-	t.Fatal(err)
-}
+s.Require().NoError(err)

@@
var foo identifier
@@
-func (s *Suite) foo(t *testing.T) {
+func (s *Suite) foo() {
  ...
 }

@@
var foo identifier
@@
-foo.AssertExpectations(t)
+foo.AssertExpectations(s.T())

# Need to change this one back
@@
@@
-    t.Cleanup(func() { mock.AssertExpectations(s.T()) })
+    t.Cleanup(func() { mock.AssertExpectations(t) })

@@
@@
import testUtil "github.com/astronomer/astro-cli/pkg/testing"

-    testUtil.MockUserInput(t, ...)
+    testUtil.MockUserInput(s.T(), ...)


# Reset the suite test back to taking testing.T
@@
var foo identifier
@@
import (
  "testing"
	"github.com/stretchr/testify/suite"
)

-func (s *Suite) foo() {
+func foo(t *testing.T) {
	  suite.Run(t, new(Suite))
 }


@@
var tt identifier
var foo identifier
@@
import (
  "testing"
	"github.com/stretchr/testify/assert"
)

func (s *Suite) foo() {
- tt.errAsseration(t, ...)
+ tt.errAsseration(s.T(), ...)
}
```

### CODE REVIEW

**Comment by kushalmalani** [comment]
@ashb This PR has been in draft state for a very long time? Do you still plan to proceed further?

**Comment by ashb** [comment]
I've just updated this PR and it's ready for review now.

**Comment by ashb** [comment]
Tests not updated by this PR

```
 M cloud/auth/auth_test.go
 M cloud/deploy/deploy_test.go
 M cloud/deployment/deployment_variable_test.go
 M cloud/deployment/inspect/inspect_test.go
 M cloud/user/user_test.go
 M cmd/airflow_test.go
 M cmd/cloud/deploy_test.go
 M cmd/cloud/deployment_inspect_test.go
 M cmd/cloud/deployment_objects_test.go
 M cmd/cloud/deployment_test.go
 M cmd/cloud/deployment_workerqueue_test.go
 M cmd/cloud/organization_test.go
 M cmd/cloud/root_test.go
 M cmd/cloud/setup_test.go
 M cmd/cloud/workspace_test.go
 M cmd/root_test.go
 M cmd/software/root_test.go
 M cmd/utils/utils_test.go
 M pkg/git/git_test.go
```

**Comment by ashb** [comment]
Yeah, almost all of these changes were automatic by the patch in the PR desc, or manually creating the driver test:


```go
type Suite struct {
	suite.Suite
}

func TestAirflowClient(t *testing.T) {
	suite.Run(t, new(Suite))
}
```

or in a few cases fixing left over state from one test to another etc.

**Comment by ashb** [comment]
Is this saying it things the coverage has dropped by 22% 😱

**Comment by ashb** [comment]
Okay I fixed that issue, but coverage is still reporting a big drop, but the individual views aren't showing what's actually dropped.

File: astro-client-core/client_test.go
Comment (ashb) [inline astro-client-core/client_test.go:1]: Oh. This file being renamed is what caused the coverage to drop. Given this test is pretty much pointless I'm going to revert these changes (they are autogenerated pkgs.)
